packages/flame_riverpod/test/widget_test.dart

```diff
@@ -136,4 +146,46 @@ void main() {
     // numberProvider as the watching component has been removed.
     expect(key.currentState?.exists(numberProvider), false);
   });
+
+  testWidgets(
+      'Test registration and de-registration of Game Provider listeners',
+      (widgetTester) async {
+    final game = MyGameWithRefAccess();
+    final key = GlobalKey<RiverpodAwareGameWidgetState>();
+
+    await widgetTester.pumpWidget(
+      ProviderScope(
+        child: MaterialApp(
+          home: RiverpodAwareGameWidget(
+            game: game,
+            key: key,
+          ),
+        ),
+      ),
+    );
+    await widgetTester.pump(Duration.zero);
+
+    // Expect the game is ready to play
+    expect(game.isAttached, true);
+    expect(game.isMounted, true);
+    expect(game.isLoaded, true);
+
+    // Pump to ensure the custom component's lifecycle events are handled
+    await widgetTester.pump(Duration.zero);
+
+    // Expect that the GameWidget is initially listening to
+    // numberProvider
+    expect(key.currentState?.exists(numberProvider), true);
+
+    // Replace the widget tree so that the GameWidget gets disposed
+    await widgetTester.pumpWidget(Container());
+    await widgetTester.pumpAndSettle();
+
+    // Expect that the component has been removed from the game.
+    expect(game.isAttached, false);
+
+    // Expect that the GameWidget is no longer listening to
+    // numberProvider as the watching component has been removed.
```

Comment:
I think the wording of this should be tweaked to reflect the comparison being made in the line.  

E.g. 

> Expect that the key no longer has access to a state.